consoles/sshVirtsh.pm

```diff
@@ -357,8 +357,9 @@ sub add_disk {
             # Avoid qemu-img's failure to get a write lock to be the reason for a job to fail
             while (1) {
                 my ($ret, $stdout, $stderr) = $self->run_cmd("qemu-img create $file $size -f qcow2", wantarray => 1);
-                die "Too many attempts to format HDD" unless ($bucket-- && $ret);
                 if ($stderr =~ /lock/i) {
+                    die "Too many attempts to format HDD" unless (--$bucket);
```

Comment:
sorry but Pre-increment operator in negated post-if operator is a bit too much obfusciation for me.

Reply:
Fixed :)

Reply:
I personally liked it (except for the useless brackets).